### PR TITLE
Use createStub whenever possible

### PR DESCRIPTION
**Pull request by greg0ire**
Just wanted to try the rector class I failed to contribute: rectorphp/rector#3042

### CODE REVIEW

File: tests/Admin/AdminTest.php [hunk 1]
```diff
@@ -1758,14 +1758,14 @@ public function testGetFilterFieldDescription(): void
 
         $modelAdmin->setModelManager($modelManager);
 
-        $pager = $this->createMock(PagerInterface::class);
+        $pager = $this->createStub(PagerInterface::class);
 
-        $datagrid = $this->createMock(DatagridInterface::class);
+        $datagrid = $this->createStub(DatagridInterface::class);
```
Comment (phansys) [inline tests/Admin/AdminTest.php:1763]: The following line calls `expects()`. Isn't that the case where the test double should be considered as a mock according the issue referenced in the Rector (https://github.com/sebastianbergmann/phpunit/issues/3120#issue-322494156)?

Comment (greg0ire) [reply tests/Admin/AdminTest.php:1763]: :thinking: well spotted… guess the rule does not work properly

Comment (VincentLanglet) [reply tests/Admin/AdminTest.php:1763]: If the tests are not failing when you expect something on a Stub, I don't really see the benefit of using it. Using both seems just confusing to me.

Comment (greg0ire) [reply tests/Admin/AdminTest.php:1763]: The tests that should fail are static analysis tests, and we don't have those for now, so indeed, let's maybe not merge this until we do.

> Using both seems just confusing to me.

I think the main goal is precisely to clear the confusion between a stub and a mock, which serve different purposes as outlined in the linked issue above. It's never been super clear to me, but I think if I start using different words, I will start noticing patterns and use them when appropriate.

Comment (greg0ire) [reply tests/Admin/AdminTest.php:1763]: I reported the bug here: https://github.com/rectorphp/rector/issues/3157

**Pull request closed without merging**